addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js

```diff
@@ -0,0 +1,322 @@
+      // BUILD
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['123.123.123.123/32', 'invalidCidr'],
+          },
+        ],
+      };
+
+      // OPERATE
+      try {
+        await service.update({}, params);
+        expect.hasAssertions();
+      } catch (err) {
+        expect(service.boom.is(err, 'badRequest')).toBe(true);
+        expect(err.message).toContain('The following are invalid CIDR inputs: invalidCidr');
+      }
+    });
+
+    it('should fail because the updateRequest contains Ipv6 CIDR ranges', async () => {
+      // BUILD
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['123.123.123.123/32', '1234::1234:abcd:ffff:1234:123/64'],
+          },
+        ],
+      };
+
+      // OPERATE
+      try {
+        await service.update({}, params);
+        expect.hasAssertions();
+      } catch (err) {
+        expect(service.boom.is(err, 'badRequest')).toBe(true);
+        expect(err.message).toContain(
+          'Currently only IPv4 formatted CIDR inputs are allowed. Please remove: 1234::1234:abcd:ffff:1234:123/64 CIDR ranges from your list',
+        );
+      }
+    });
+
+    it('should fail because the updateRequest contains duplicate protocol-port combinations', async () => {
+      // BUILD
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['123.123.123.123/32'],
+          },
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['456.456.456.456/32'],
+          },
+        ],
+      };
+
+      // OPERATE
+      try {
+        await service.update({}, params);
+        expect.hasAssertions();
+      } catch (err) {
+        expect(service.boom.is(err, 'badRequest')).toBe(true);
+        expect(err.message).toContain(
+          'The update request contains duplicate protocol-port combinations. Please make sure all ingress rule objects have unique protocol-fromPort-toPort combinations',
+        );
+      }
+    });
+  });
+
+  describe('update', () => {
+    it('should fail because the user is not authorized', async () => {
+      // BUILD
+      const requestContext = {};
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['123.123.123.123/32'],
+          },
+        ],
+      };
+      service.assertAuthorized.mockImplementationOnce(() => {
+        throw new Error('User is not authorized');
+      });
```

Comment:
This here looks good to me. Is it possible to add a couple of unit tests in auth module as well?

Reply:
Working on it..

Reply:
Done. Added those pertaining to the CIDR update switch-case.